go-mode.el

```diff
@@ -606,31 +644,36 @@ current line will be returned."
        ((looking-at "[])}]")
         (go-goto-opening-parenthesis)
         (if (and
-             (not (eq (char-after) ?\()) ; opening parens always indent
+             (not (eq (char-after) ?\())
              (go-previous-line-has-dangling-op-p))
-            (- (current-indentation) tab-width)
+            (go--non-dangling-indent)
           (go--indentation-for-opening-parenthesis)))
-       ((progn (go--backward-irrelevant t)
-               (looking-back go-dangling-operators-regexp
-                             (- (point) go--max-dangling-operator-length)))
+       ((let ((dangling-line (go-previous-line-has-dangling-op-p)))
+         (when dangling-line (goto-char dangling-line) (end-of-line) t))
```

Comment:
This (let ... (when ... (point movement))) in the cond condition is fairly hard to follow and a bit confusing why we're doing extra work in the condition unrelated to whether it evaluates to true or not. 

Maybe move the let declaration up out of the (cond) and move the movement stuff into the body of this condition?

Reply:
Done.